Updated login/registration styling

gsidebo opened the pull request:
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
  - [x] Mobile width screenshots
  - [x] Tag @ferdi or @pdpinch for review
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Closes #263 

#### What's this PR do?
- Updates the login and signup UI/styling
- Adds a notification when a user enters their email to sign up and submits

#### How should this be manually tested?
- Go through the login and signup flows, make sure the UI looks good and matches the designs
- Make sure the "email sent" notification in the signup flow behaves as expected 

#### Screenshots (if appropriate)
<img width="955" alt="ss 2019-05-21 at 15 17 52 " src="https://user-images.githubusercontent.com/14932219/58124229-bd919600-7bdb-11e9-8a2f-8785350f35ed.png">
<img width="396" alt="ss 2019-05-21 at 15 18 10 " src="https://user-images.githubusercontent.com/14932219/58124230-bd919600-7bdb-11e9-9da0-7ed8c2101251.png">
<img width="951" alt="ss 2019-05-21 at 12 32 43 " src="https://user-images.githubusercontent.com/14932219/58124258-d39f5680-7bdb-11e9-9308-1bd5f29a8d4e.png">
<img width="947" alt="ss 2019-05-21 at 12 32 59 " src="https://user-images.githubusercontent.com/14932219/58124259-d39f5680-7bdb-11e9-90f3-a370b8d14a2e.png">
<img width="369" alt="ss 2019-05-21 at 13 09 20 " src="https://user-images.githubusercontent.com/14932219/58124300-eb76da80-7bdb-11e9-8f88-03969c70a29a.png">
<img width="1134" alt="ss 2019-05-21 at 12 53 23 " src="https://user-images.githubusercontent.com/14932219/58124310-efa2f800-7bdb-11e9-9c51-008cf0643129.png">
<img width="358" alt="ss 2019-05-21 at 12 53 34 " src="https://user-images.githubusercontent.com/14932219/58124326-f29de880-7bdb-11e9-9fe9-357bddd9302a.png">

gsidebo commented:
@pdpinch @Ferdi @abdulkdawson Please review the screenshots ^^^

abdulkdawson commented:
@gsidebo we need to change the header text sizing for mobile. Its a bit too big. Whats the mobile screen size that you're showing?

Ferdi commented:
@gsidebo  let me know if it's out of scope but can we keep the footer down the page even when there is not much content on the page ?

gsidebo commented:
@Ferdi I created a separate issue to address that https://github.com/mitodl/mitxpro/issues/333

rhysyngsun commented on static/js/components/forms/RegisterEmailForm.js:
```diff
@@ -29,25 +29,31 @@ const RegisterEmailForm = ({ onSubmit }: Props) => (
               onRecaptcha={value => setFieldValue("recaptcha", value)}
               recaptchaKey={SETTINGS.recaptchaKey}
             />
             <ErrorMessage name="recaptcha" component="div" />
```
Could you add `component={FormError}` to the recaptcha error so it gets the same treatment?

gsidebo commented:
@rhysyngsun 2 bits of FYI

1. I changed the signup form to redirect the user to the login page after the email:
    
    ![ss 2019-05-22 at 12 54 58 ](https://user-images.githubusercontent.com/14932219/58193416-ede73c00-7c90-11e9-9ad9-cfe33eda56a3.png)
    Displaying the "email sent" notification and showing the home page/course catalog (both non-React pages) would require something novel. Doesn't seem important enough to figure that out now, so I'm hoping the `/login` redirect will suffice. @Ferdi let me know what you think about that
1. I changed up the password validation a bit. I now have separate validators for the login password form (i.e.: existing passwords) and the register form. New passwords will check for the letter/number combo while the validator for existing passwords will not. I just did this to make things a bit more developer-friendly since we created users/passwords before this constraint was added, and I don't think there's any risk. If you can think of some risk I'm not seeing, lmk

Ferdi commented:
> Displaying the "email sent" notification and showing the home page/course catalog (both non-React pages) would require something novel. Doesn't seem important enough to figure that out now, so I'm hoping the /login redirect will suffice

OK

Does registration respects any "next" param ?

gsidebo commented:
@Ferdi looks like it doesn't right now. Did you have a use case in mind?

rhysyngsun commented:
@gsidebo @Ferdi it does

https://github.com/mitodl/mitxpro/blob/master/static/js/containers/pages/register/RegisterEmailPage.js#L42
